src/Raven.Pal/src/shared_all.c

```diff
@@ -0,0 +1,73 @@
+#include <sys/types.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "rvn.h"
+#include "rvn_internal.h"
+#include "status_codes.h"
+
+
+void noop_void(int64_t size, char* filename){}
+bool noop_bool(int64_t size, char* filename){ return false; }
+
+
+MemoryLockCallback g_locked_memory_callback = noop_void;
+RecoveryMemoryLockFailureCallback g_recovery_memory_lock_failure_callback = noop_bool;
+
+EXPORT
+void rvn_register_callbacks(MemoryLockCallback memoryLockCallback, 
+    RecoveryMemoryLockFailureCallback recoveryMemoryLockFailureCallback)
+{
+    g_locked_memory_callback = memoryLockCallback;
+    g_recovery_memory_lock_failure_callback = recoveryMemoryLockFailureCallback;
+}
+
+PRIVATE int64_t
+_nearest_size_to_page_size(int64_t orig_size, int64_t sys_page_size)
+{
+    int64_t mod = orig_size % sys_page_size;
+    if (mod == 0)
+    {
+        return orig_size;
+    }
+    return ((orig_size / sys_page_size) + 1) * sys_page_size;
+}
+
+rvn_write_mode _get_writer_mode()
+{
+    const char* env_var_name = "RAVEN_WRITER_MODE";
```

Comment:
RAVEN_VORON_WRITER_MODE

Reply:
We aren't using the same approach, but I think that using `RAVEN_STORAGE_WRITE_MODE` might actually be a better match for our needs, no? Same as our usual env variables.
But we don't provide another way to configure it, so maybe we _want_ to be different?

Reply:
this is how we are calling envs elsewhere e.g. https://github.com/ravendb/ravendb/blob/9b9d481552d537f2e2173a75768a3d694661e43c/src/Raven.Client/Documents/Conventions/DocumentConventions.cs#L199
I think we should keep them consistent

Reply:
Agree